### PR TITLE
[amp-story] Ensure clicks on anchor tags are handled by tooltip 🐛

### PR DESCRIPTION
**Pull request by Enriqe**
Currently, if a publisher specifies the following markup:
`
<a href="google.com"> 
  <span> Click me! </span>
</a>
`

The click will not be handled by the tooltip code since the target will be the `<span>` element, and we currently are only handling `<a>` tags.

This PR ensures that anything inside `<a>` tags won't be clickable and that target triggering the click event will be the `<a>` tag.

Other proposals considered:
1. Add a listener on each <a> with capture: true, and use the currentTarget instead of target.
2. Check if the target is descendant of <a>.
3. Add a click shield on <a> using pseudo-element.

(1) Might interfere down the line with other event listeners.
(2) Might be costly to do on every click (especially when the DOM is deep). 
(3) Is not possible without overwriting the position of the `<a>` tag (to add a pseudo selector we need the `<a>` tag to have either `position: relative` or `position:absolute`.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story.css [hunk 1]
```diff
@@ -321,6 +321,12 @@ amp-story-grid-layer > * {
   pointer-events: auto !important;
 }
 
+/** Ensure clicks inside anchor tags are handled by the tooltip. */
+amp-story-grid-layer a * {
+  pointer-events: none;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.css]: `!important`

File: extensions/amp-story/1.0/amp-story.css [hunk 2]
```diff
@@ -321,6 +321,12 @@ amp-story-grid-layer > * {
...
+  cursor: default;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.css]: Since these don't have pointer events, does the cursor property actually do anything?

**Comment by gmajoulet** [comment]
Could there be <a> links in the page we're not thinking of? Like links within another component?

**Comment by Enriqe** [comment]
Not under the `amp-story-grid-layer`. The other components they are allowed in are in a shadow DOM or on a different kind of layer (e.g. cta-layer, system layer, bookend, attachment...).

**Comment by gmajoulet** [comment]
Cool, thanks. Should we reset the pointer events on links within an embedded component? They should be contained in an iframe, but idk if all components will be? You have more context on this, so your call : )

**Comment by Enriqe** [comment]
After some sanity checking it looks like the embedded component handles them without problem, even with this new rule. But I will keep an eye on it, especially because I am changing related stuff on my other PR :)

**Comment by newmuis** [comment]
Can we generalize this to some clickable element classname? We would also want this behavior for non-link clickable elements, right?

**Comment by Enriqe** [comment]
For the moment it would only be link elements, as the other embedded components would still require a click shield (so that the click doesn't get consumed by the iframe). In which case I think adding a classname only for links sounds like some overhead that doesn't bring much value, WDYT? @newmuis

**Pull request merged**